store/tikv/region_cache.go

```diff
@@ -888,10 +888,11 @@ func (c *RegionCache) LoadRegionsInKeyRange(bo *Backoffer, startKey, endKey []by
 		}
 		regions = append(regions, batchRegions...)
 		endRegion := batchRegions[len(batchRegions)-1]
-		if endRegion.ContainsByEnd(endKey) {
+		regionEndKey := endRegion.EndKey()
+		if endRegion.ContainsByEnd(endKey) || bytes.Equal(regionEndKey, endKey) {
```

Comment:
If `bytes.Equal(regionEndKey, endKey)`, `endRegion.ContainsByEnd(endKey)` must be true.

Reply:
Sorry, I tested with 4.0.0, without noticing that the bug has already been fixed in newer version. PR will be closed.